test/integration/targets/get_url/tasks/main.yml

```diff
@@ -65,6 +65,28 @@
     that:
         - result.failed
 
+- name: test HTTP HEAD request for file in check mode
+  get_url: url="http://{{ httpbin_host }}/get" dest={{output_dir}}/get_url_check.txt force=yes
+  check_mode: True
+  register: result
+
+- name: assert that the HEAD request was successful in check mode
+  assert:
+    that:
+    - result.changed
+    - '"OK" in result.msg'
+
+- name: test HTTP HEAD for nonexistent URL in check mode
+  get_url: url="http://brokenurl.test/missingfile.test" dest={{output_dir}}/shouldnotexist.html force=yes
```

Comment:
@sivel You're the httptester expert. Any suggestion for this test?

@SesquipedalianDefenestrator We should keep external dependencies minimal, so there should be a httptester solution or something local.

Probably http://{{httpbin_host}/DOESNOTEXIST is a good-enough URL.